### ISSUE TITLE
Contributors are able to upload via Gallery block

### ISSUE THREAD

**Issue by SiobhyB** [wordpress-mobile/gutenberg-mobile#4293]
**Describe the bug**

It shouldn't be possible for users with the contributor role to upload media to a site, but it's currently possible for them to do so via the Gallery block in the Android and iOS apps. 

As reported separately in https://github.com/wordpress-mobile/gutenberg-mobile/issues/3259, it also _appears_ like contributors are able to upload media via the Image block, up until they see an error message following the upload. However, with the Gallery block, contributors actually _can_ fully upload media to a site at the moment. 

**To Reproduce**

1. Log into an account where you're listed as a contributor on a site.
2. Navigate to the editor and add the Gallery block.
3. Go through the steps to upload some images to the newly added Gallery block.
4. Save and submit your changes.
5. Next log into the account where you're an admin on the test site to confirm the post has been submitted with the uploaded media in the Gallery.

**Expected behavior**

After adding a Gallery block while logged in as a contributor on the web, I see the following error informing me of the limitations:

<img width="914" alt="Screenshot 2021-11-25 at 18 06 27" src="https://user-images.githubusercontent.com/2998162/143487150-bf1c21ba-ed0d-46b0-801e-37f9d8df94ca.png">

I'd expect similar behaviour in the apps, instead of actually being able to upload images. 

**Screenshots**

Below, you can see a screenshot of the app's editor, where I was able to upload images to a Gallery block with no error message. The second screenshot is my view from an admin account on the web, where it's confirmed the images were uploaded by the contributor:

| Contributor  | Admin |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/143489390-d5ba156d-0b35-4bb1-82da-984002d6e309.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/143489520-81ea54b7-ff7b-41f2-8785-872b2d6b23c5.png" width="100%"> |

**Smartphone (please complete the following information):**
 - Device: Pixel 5 and iPhone XR
 - OS: Android 12 and iOS 14.6
 - Version: Tested on 18.6 and 18.7 (alpha)

**Additional context**

There is an older related issue for Android here that's now closed, but may be a helpful reference: https://github.com/wordpress-mobile/WordPress-Android/issues/10666

In addition, the same bug can be seen in the Tiled Gallery block, which is currently in progress.

**Comment by guarani** [comment]
The Tiled Gallery team haven't prioritized this because it's an existing but that affects the core Gallery and is on the Mobile Gutenpages project board.

**Comment by mchowning** [comment]
👋  @illusaen ! @hypest  asked me to take a look at this, so I'll reassign the issue to me to make sure we don't duplicate efforts.

**Comment by mchowning** [comment]
I'm seeing an error on both WPiOS and WPAndroid when I try to upload an image from a contributor account to a gallery block. Similarly, I see an upload failure with the Tiled Gallery block. 

It sounds like that's not what you were seeing here @SiobhyB ? Maybe something fixed this since you filed the issue?

https://user-images.githubusercontent.com/4656348/160168484-6db78112-25e0-49ec-9914-75f97fc2b803.mp4

I'm have a fix for https://github.com/wordpress-mobile/gutenberg-mobile/issues/3259 that will remove the upload options from media blocks for contributors, which will obviously improve things here.

**Comment by SiobhyB** [comment]
@mchowning, I re-tested and am seeing an error when attempting to upload via a contributor account now, too. 🎉 

I'm fairly certain I was able to upload as a contributor at the time I published this issue, as I remember testing a few times to confirm if this was an issue specific to the Tiled Gallery or also the core Gallery. The evidence indicates I made a mistake, though, as I can't even reproduce with 18.6 (the version referenced in this issue). Strangely, the older, related issue I dug up in https://github.com/wordpress-mobile/WordPress-Android/issues/10666 mentions a couple of unexplained cases of the permissions changing in the comments, so perhaps there's a small chance of something else at play here. 🤔 

Either way, it seems like removing the options altogether will prevent anything like this happening in the future! :) I'll go ahead to close this issue as we're not able to reproduce.
